pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test: TestTenantLogicCCL_cluster_locks_tenant failed

pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test.TestTenantLogicCCL_cluster_locks_tenant [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsGcpLinuxX8664BigVm_CclUnitTests/14626616?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsGcpLinuxX8664BigVm_CclUnitTests/14626616?buildTab=artifacts#/) on master @ [2a5e231716c436781f12452d800651f51c6383b7](https://github.com/cockroachdb/cockroach/commits/2a5e231716c436781f12452d800651f51c6383b7):


```
=== RUN   TestTenantLogicCCL_cluster_locks_tenant
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/020c62f143b22628b86ce9738a03dba8/logTestTenantLogicCCL_cluster_locks_tenant1701138771
    test_log_scope.go:81: use -show-logs to present logs inline
    logic.go:3021: let $root_session = 17c17635ff3679170000000000000001
    logic.go:3021: let $testuser_session = 17c1763622ff33360000000000000001
    logic.go:3021: let $txn1 = e37b30d8-6b9b-4355-b2e3-f1242d0b0e2d
    logic.go:3021: let $txn2 = af298c7a-c371-44ca-bdf4-40074d3c0821
    logic.go:3021: let $txn3 = 11a97fb2-2348-4597-947e-6fafa3105ac2
    logic.go:3021: let $txn4 = ed672888-de7e-4a43-9b49-0c5d4e7f2482
[06:15:11] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/5446/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/cluster_locks_tenant: 26 statements
    logic.go:2963: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/5446/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/cluster_locks_tenant:230: SELECT count(*) FROM crdb_internal.cluster_locks WHERE table_name IN ('t','t2')
        expected:
            4
            
        but found (query options: "retry") :
            0
            
[06:15:57] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/5446/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/cluster_locks_tenant: 36 statements
[06:15:57] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/5446/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/cluster_locks_tenant with config 3node-tenant: 36 tests, 1 failures
    logic.go:4145: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/5446/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/cluster_locks_tenant:235: error while processing
    logic.go:4145: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/5446/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/cluster_locks_tenant:235: too many errors encountered, skipping the rest of the input
    panic.go:626: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/020c62f143b22628b86ce9738a03dba8/logTestTenantLogicCCL_cluster_locks_tenant1701138771
--- FAIL: TestTenantLogicCCL_cluster_locks_tenant (55.03s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantLogicCCL_cluster_locks_tenant.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-37224

pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test.TestTenantLogicCCL_cluster_locks_tenant [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsGcpLinuxX8664BigVm_CclUnitTests/14635713?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsGcpLinuxX8664BigVm_CclUnitTests/14635713?buildTab=artifacts#/) on master @ [7fc4c7bcbbf0c75a62d056da0bf79a5a32714650](https://github.com/cockroachdb/cockroach/commits/7fc4c7bcbbf0c75a62d056da0bf79a5a32714650):


```
=== RUN   TestTenantLogicCCL_cluster_locks_tenant
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/020c62f143b22628b86ce9738a03dba8/logTestTenantLogicCCL_cluster_locks_tenant4275148946
    test_log_scope.go:81: use -show-logs to present logs inline
    logic.go:3021: let $root_session = 17c1c50f030a07cc0000000000000001
    logic.go:3021: let $testuser_session = 17c1c50f260e0a6c0000000000000001
    logic.go:3021: let $txn1 = 0b46de9d-c696-424b-b1f2-7b323709065a
    logic.go:3021: let $txn2 = ffea5e69-0a64-4234-9e5c-9b63a80a70f8
    logic.go:3021: let $txn3 = e5dcb3c7-ad20-44db-a97b-0ff6dd33a973
    logic.go:3021: let $txn4 = 1b59198f-e363-4836-b0c5-427b9cf128a3
[06:20:05] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/5519/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/cluster_locks_tenant: 26 statements
[06:20:05] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/5519/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/cluster_locks_tenant with config 3node-tenant: 30 tests, 0 failures
    logic.go:4145: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/5519/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/cluster_locks_tenant:204: error while processing
    logic.go:4145: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/5519/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/cluster_locks_tenant:204: 
        expected success, but found
        (40001) restart transaction: TransactionRetryWithProtoRefreshError: TransactionRetryError: retry txn (RETRY_ASYNC_WRITE_FAILURE - missing intent on: /Tenant/10/Table/106/1/"c"/0): "sql txn" meta={id=1b59198f key=/Tenant/10/Table/106/1/"b" iso=Serializable pri=0.00577965 epo=0 ts=1711866002.430824821,2 min=1711866002.385191052,0 seq=2} lock=true stat=PENDING rts=1711866002.430824821,2 wto=false gul=1711866002.885191052,0
    panic.go:626: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/020c62f143b22628b86ce9738a03dba8/logTestTenantLogicCCL_cluster_locks_tenant4275148946
--- FAIL: TestTenantLogicCCL_cluster_locks_tenant (8.79s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #121410 pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test: TestTenantLogicCCL_cluster_locks_tenant failed [C-test-failure O-robot T-sql-queries branch-release-24.1 release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantLogicCCL_cluster_locks_tenant.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test.TestTenantLogicCCL_cluster_locks_tenant [failed](https://tanzanite.cluster.engflow.com/invocations/default/a59f8357-e13f-41ac-b531-f060fa4bcba4?testReportRun=7&testReportShard=16&testReportAttempt=1#targets-Ly9wa2cvY2NsL2xvZ2ljdGVzdGNjbC90ZXN0cy8zbm9kZS10ZW5hbnQ6M25vZGUtdGVuYW50X3Rlc3Q=) on master @ [7fc4c7bcbbf0c75a62d056da0bf79a5a32714650](https://github.com/cockroachdb/cockroach/commits/7fc4c7bcbbf0c75a62d056da0bf79a5a32714650):


```
=== RUN   TestTenantLogicCCL_cluster_locks_tenant
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestTenantLogicCCL_cluster_locks_tenant259184025
    test_log_scope.go:81: use -show-logs to present logs inline
[04:15:38] setting distsql_workmem='23626B';
[04:15:38] setting distsql_workmem='23626B';
    logic.go:3021: let $root_session = 17c20cd9df595b370000000000000001
    logic.go:3021: let $testuser_session = 17c20cd9f6bfd5370000000000000001
    logic.go:3021: let $txn1 = a7649597-8f42-473a-ac94-60c37cc799b7
    logic.go:3021: let $txn2 = 6d40ae60-a18e-475d-8182-6d51d927ba64
    logic.go:3021: let $txn3 = e9976108-b378-4808-90d8-1b8398e9463d
    logic.go:3021: let $txn4 = 02314e22-9223-43a0-afa6-3bd2fc6b268d
[04:15:41] --- progress: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/cluster_locks_tenant: 26 statements
[04:15:41] --- done: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/cluster_locks_tenant with config 3node-tenant: 30 tests, 0 failures
    logic.go:4145: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/cluster_locks_tenant:204: error while processing
    logic.go:4145: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/cluster_locks_tenant:204: 
        expected success, but found
        (40001) restart transaction: TransactionRetryWithProtoRefreshError: TransactionRetryError: retry txn (RETRY_ASYNC_WRITE_FAILURE - missing intent on: /Tenant/10/Table/106/1/"c"/0): "sql txn" meta={id=02314e22 key=/Tenant/10/Table/106/1/"b" iso=Serializable pri=0.00223224 epo=0 ts=1711944938.766138890,2 min=1711944938.734692448,0 seq=2} lock=true stat=PENDING rts=1711944938.766138890,2 wto=false gul=1711944939.234692448,0
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestTenantLogicCCL_cluster_locks_tenant259184025
--- FAIL: TestTenantLogicCCL_cluster_locks_tenant (8.27s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=7</code>
 - <code>shard=16</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #121410 pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test: TestTenantLogicCCL_cluster_locks_tenant failed [C-test-failure O-robot T-sql-queries branch-release-24.1 release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantLogicCCL_cluster_locks_tenant.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test.TestTenantLogicCCL_cluster_locks_tenant [failed](https://tanzanite.cluster.engflow.com/invocations/default/62b38762-ec59-46b4-86f6-d7d983206348?testReportRun=11&testReportShard=16&testReportAttempt=1#targets-Ly9wa2cvY2NsL2xvZ2ljdGVzdGNjbC90ZXN0cy8zbm9kZS10ZW5hbnQ6M25vZGUtdGVuYW50X3Rlc3Q=) on master @ [c43f54cdde5b7578f4a0ca61de41463f0d690993](https://github.com/cockroachdb/cockroach/commits/c43f54cdde5b7578f4a0ca61de41463f0d690993):


```
=== RUN   TestTenantLogicCCL_cluster_locks_tenant
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestTenantLogicCCL_cluster_locks_tenant1076918167
    test_log_scope.go:81: use -show-logs to present logs inline
[05:09:21] setting distsql_workmem='23468B';
[05:09:21] setting distsql_workmem='23468B';
    logic.go:3021: let $root_session = 17c25e5cec51c6c70000000000000001
    logic.go:3021: let $testuser_session = 17c25e5d0477e1bf0000000000000001
    logic.go:3021: let $txn1 = 970b50d4-b3df-4224-a308-7f92b546807f
    logic.go:3021: let $txn2 = e3a198fd-4b64-4c83-b114-fdeb2568317e
    logic.go:3021: let $txn3 = 435069cf-34fb-4e51-815a-a96e09bc4913
    logic.go:3021: let $txn4 = 5a4965c2-77f8-4270-a5f3-1dbaf29b808b
[05:09:25] --- progress: /var/lib/engflow/worker/work/2/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/cluster_locks_tenant: 26 statements
[05:09:25] --- done: /var/lib/engflow/worker/work/2/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/cluster_locks_tenant with config 3node-tenant: 30 tests, 0 failures
    logic.go:4145: 
        /var/lib/engflow/worker/work/2/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/cluster_locks_tenant:204: error while processing
    logic.go:4145: 
        /var/lib/engflow/worker/work/2/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/cluster_locks_tenant:204: 
        expected success, but found
        (40001) restart transaction: TransactionRetryWithProtoRefreshError: TransactionRetryError: retry txn (RETRY_ASYNC_WRITE_FAILURE - missing intent on: /Tenant/10/Table/106/1/"c"/0): "sql txn" meta={id=5a4965c2 key=/Tenant/10/Table/106/1/"b" iso=Serializable pri=0.00447482 epo=0 ts=1712034562.070819290,2 min=1712034562.039439716,0 seq=2} lock=true stat=PENDING rts=1712034562.070819290,2 wto=false gul=1712034562.539439716,0
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestTenantLogicCCL_cluster_locks_tenant1076918167
--- FAIL: TestTenantLogicCCL_cluster_locks_tenant (7.70s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=11</code>
 - <code>shard=16</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantLogicCCL_cluster_locks_tenant.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test.TestTenantLogicCCL_cluster_locks_tenant [failed](https://tanzanite.cluster.engflow.com/invocations/default/6c388c68-b7e8-4c4f-b47d-159a54ff1853?testReportRun=6&testReportShard=16&testReportAttempt=1#targets-Ly9wa2cvY2NsL2xvZ2ljdGVzdGNjbC90ZXN0cy8zbm9kZS10ZW5hbnQ6M25vZGUtdGVuYW50X3Rlc3Q=) on master @ [7166724ff02b9b7f0d5787f74d919487ca347bd0](https://github.com/cockroachdb/cockroach/commits/7166724ff02b9b7f0d5787f74d919487ca347bd0):


```
=== RUN   TestTenantLogicCCL_cluster_locks_tenant
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestTenantLogicCCL_cluster_locks_tenant3785568505
    test_log_scope.go:81: use -show-logs to present logs inline
[06:24:06] setting distsql_workmem='66419B';
[06:24:06] setting distsql_workmem='66419B';
    logic.go:3021: let $root_session = 17c2b105b237b51e0000000000000001
    logic.go:3021: let $testuser_session = 17c2b105cab92c5d0000000000000001
    logic.go:3021: let $txn1 = c7fb482b-0155-4ffd-984f-5e7758297a3d
    logic.go:3021: let $txn2 = 9a72223b-4abd-4ed8-9aa9-7b7e65a7842b
    logic.go:3021: let $txn3 = 445a0884-bd85-40d9-aedb-f8f71f3a8bec
    logic.go:3021: let $txn4 = 6f6757ab-acd3-4771-90cc-ecf0521b670f
[06:24:10] --- progress: /var/lib/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/cluster_locks_tenant: 26 statements
[06:24:10] --- done: /var/lib/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/cluster_locks_tenant with config 3node-tenant: 30 tests, 0 failures
    logic.go:4145: 
        /var/lib/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/cluster_locks_tenant:204: error while processing
    logic.go:4145: 
        /var/lib/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/cluster_locks_tenant:204: 
        expected success, but found
        (40001) restart transaction: TransactionRetryWithProtoRefreshError: TransactionRetryError: retry txn (RETRY_ASYNC_WRITE_FAILURE - missing intent on: /Tenant/10/Table/106/1/"c"/0): "sql txn" meta={id=6f6757ab key=/Tenant/10/Table/106/1/"b" iso=Serializable pri=0.00299485 epo=0 ts=1712125446.904616363,2 min=1712125446.876344397,0 seq=2} lock=true stat=PENDING rts=1712125446.904616363,2 wto=false gul=1712125447.376344397,0
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestTenantLogicCCL_cluster_locks_tenant3785568505
--- FAIL: TestTenantLogicCCL_cluster_locks_tenant (7.85s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=6</code>
 - <code>shard=16</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantLogicCCL_cluster_locks_tenant.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test.TestTenantLogicCCL_cluster_locks_tenant [failed](https://tanzanite.cluster.engflow.com/invocations/default/5b65eb98-ea00-4935-9667-b112196a8e83?testReportRun=7&testReportShard=16&testReportAttempt=1#targets-Ly9wa2cvY2NsL2xvZ2ljdGVzdGNjbC90ZXN0cy8zbm9kZS10ZW5hbnQ6M25vZGUtdGVuYW50X3Rlc3Q=) on master @ [0d3c5b71415de867fedde97de01486d424209848](https://github.com/cockroachdb/cockroach/commits/0d3c5b71415de867fedde97de01486d424209848):


```
=== RUN   TestTenantLogicCCL_cluster_locks_tenant
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestTenantLogicCCL_cluster_locks_tenant3050186744
    test_log_scope.go:81: use -show-logs to present logs inline
[06:12:32] setting distsql_workmem='26067B';
[06:12:32] setting distsql_workmem='26067B';
    logic.go:3021: let $root_session = 17c2fef89e2a50bf0000000000000001
    logic.go:3021: let $testuser_session = 17c2fef8b6afa8790000000000000001
    logic.go:3021: let $txn1 = 1feda179-d311-4b87-8565-b6509653544d
    logic.go:3021: let $txn2 = ad6a66e7-60f8-466b-ba6d-6b7c2f02607c
    logic.go:3021: let $txn3 = c5aa6d79-7520-45bd-a830-4e55dd84d1ff
    logic.go:3021: let $txn4 = f71bdc25-df23-4b8e-b5c2-12c6b888379e
[06:12:35] --- progress: /var/lib/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/cluster_locks_tenant: 26 statements
[06:12:35] --- done: /var/lib/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/cluster_locks_tenant with config 3node-tenant: 30 tests, 0 failures
    logic.go:4148: 
        /var/lib/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/cluster_locks_tenant:204: error while processing
    logic.go:4148: 
        /var/lib/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/cluster_locks_tenant:204: 
        expected success, but found
        (40001) restart transaction: TransactionRetryWithProtoRefreshError: TransactionRetryError: retry txn (RETRY_ASYNC_WRITE_FAILURE - missing intent on: /Tenant/10/Table/106/1/"c"/0): "sql txn" meta={id=f71bdc25 key=/Tenant/10/Table/106/1/"b" iso=Serializable pri=0.01998693 epo=0 ts=1712211152.650190821,2 min=1712211152.619659444,0 seq=2} lock=true stat=PENDING rts=1712211152.650190821,2 wto=false gul=1712211153.119659444,0
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestTenantLogicCCL_cluster_locks_tenant3050186744
--- FAIL: TestTenantLogicCCL_cluster_locks_tenant (8.06s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=7</code>
 - <code>shard=16</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantLogicCCL_cluster_locks_tenant.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

--- a/pkg/ccl/logictestccl/testdata/logic_test/cluster_locks_tenant
+++ b/pkg/ccl/logictestccl/testdata/logic_test/cluster_locks_tenant
@@ -179,20 +179,30 @@ user_name  query                                         phase
 testuser   DELETE FROM t WHERE (k >= 'b') AND (k < 'x')  executing
 
 # looking at each transaction separately, validate the expected results in the lock table
-query TTTTTTTBB colnames,retry
+query TTTTTTTBB colnames,rowsort,retry
 SELECT database_name, schema_name, table_name, lock_key_pretty, lock_strength, durability, isolation_level, granted, contended FROM crdb_internal.cluster_locks WHERE table_name='t' AND txn_id='$txn3'
 ----
-database_name  schema_name  table_name  lock_key_pretty  lock_strength  durability  isolation_level  granted  contended
+database_name  schema_name  table_name  lock_key_pretty     lock_strength  durability    isolation_level  granted  contended
+test           public       t           /Table/106/1/"a"/0  Exclusive      Unreplicated  SERIALIZABLE     true     false
+test           public       t           /Table/106/1/"b"/0  Exclusive      Unreplicated  SERIALIZABLE     true     true
+test           public       t           /Table/106/1/"c"/0  Exclusive      Unreplicated  SERIALIZABLE     true     false
+test           public       t           /Table/106/1/"l"/0  Exclusive      Unreplicated  SERIALIZABLE     true     false
+test           public       t           /Table/106/1/"m"/0  Exclusive      Unreplicated  SERIALIZABLE     true     false
+test           public       t           /Table/106/1/"p"/0  Exclusive      Unreplicated  SERIALIZABLE     true     false
+test           public       t           /Table/106/1/"s"/0  Exclusive      Unreplicated  SERIALIZABLE     true     false
+test           public       t           /Table/106/1/"t"/0  Exclusive      Unreplicated  SERIALIZABLE     true     false
+test           public       t           /Table/106/1/"z"/0  Exclusive      Unreplicated  SERIALIZABLE     true     false
 
-query TTTTTTTBB colnames
+query TTTTTTTBB colnames,rowsort
 SELECT database_name, schema_name, table_name, lock_key_pretty, lock_strength, durability, isolation_level, granted, contended FROM crdb_internal.cluster_locks WHERE table_name='t' AND txn_id='$txn4'
 ----
-database_name  schema_name  table_name  lock_key_pretty  lock_strength  durability  isolation_level  granted  contended
+database_name  schema_name  table_name  lock_key_pretty     lock_strength  durability    isolation_level  granted  contended
+test           public       t           /Table/106/1/"b"/0  Exclusive      Unreplicated  SERIALIZABLE     false    true
 
 query I
 SELECT count(*) FROM crdb_internal.cluster_locks WHERE table_name = 't'
 ----
-0
+10
 
 statement ok
 ROLLBACK

--- a/pkg/sql/logictest/logic.go
+++ b/pkg/sql/logictest/logic.go
@@ -1489,16 +1489,6 @@ func (t *logicTest) newCluster(
 	setSQLTestingKnobs(&params.ServerArgs.Knobs)
 
 	cfg := t.cfg
-	if cfg.UseSecondaryTenant == logictestbase.Always {
-		// In the tenant case we need to enable replication in order to split and
-		// relocate ranges correctly.
-		//
-		// TODO(#76378): This condition is faulty. In the case where the
-		// profile is configured with "Random", we want to set the
-		// replication mode as well when a test tenant is effectively
-		// created. This currently is not happening.
-		params.ReplicationMode = base.ReplicationAuto
-	}
 	if cfg.BootstrapVersion != clusterversion.Key(0) {
 		if params.ServerArgs.Knobs.Server == nil {
 			params.ServerArgs.Knobs.Server = &server.TestingKnobs{}

--- a/pkg/sql/logictest/testdata/logic_test/distsql_tenant_locality
+++ b/pkg/sql/logictest/testdata/logic_test/distsql_tenant_locality
@@ -1,20 +1,29 @@
 # LogicTest: 3node-tenant-multiregion
 
+user host-cluster-root
+
+statement ok
+ALTER TENANT [10] GRANT CAPABILITY can_admin_relocate_range=true
+
+user root
+
 # Create a table on the secondary tenant.
 statement ok
 CREATE TABLE t (k INT PRIMARY KEY, v INT, FAMILY (k, v));
 INSERT INTO t SELECT i, i FROM generate_series(1, 6) AS g(i)
 
+# Upreplicate the table's range.
+statement ok
+ALTER TABLE t EXPERIMENTAL_RELOCATE VALUES (ARRAY[1, 2, 3], 0)
+
 # Split the ranges in the table.
 statement ok
 ALTER TABLE t SPLIT AT SELECT generate_series(1, 6)
 
-# Relocate ranges in the admin tenant so that
+# Relocate ranges so that
 # - [1-2) and [2-3) are on node 2
 # - [3-4) and [4-5) are on node 3
 # - [5-6) and [6-7) are on node 1.
-user host-cluster-root
-
 statement ok
 ALTER RANGE RELOCATE LEASE TO 2 FOR SELECT range_id FROM crdb_internal.ranges WHERE start_pretty LIKE '%Tenant%1';
 ALTER RANGE RELOCATE LEASE TO 2 FOR SELECT range_id FROM crdb_internal.ranges WHERE start_pretty LIKE '%Tenant%2';
@@ -22,8 +31,6 @@ ALTER RANGE RELOCATE LEASE TO 3 FOR SELECT range_id FROM crdb_internal.ranges WH
 ALTER RANGE RELOCATE LEASE TO 3 FOR SELECT range_id FROM crdb_internal.ranges WHERE start_pretty LIKE '%Tenant%4';
 ALTER RANGE RELOCATE LEASE TO 1 FOR SELECT range_id FROM crdb_internal.ranges WHERE start_pretty LIKE '%Tenant%5';
 ALTER RANGE RELOCATE LEASE TO 1 FOR SELECT range_id FROM crdb_internal.ranges WHERE start_pretty LIKE '%Tenant%6';
-
-user root
 
 # Populate the range cache.
 statement ok

--- a/pkg/sql/opt/exec/execbuilder/testdata/distsql_tenant_locality
+++ b/pkg/sql/opt/exec/execbuilder/testdata/distsql_tenant_locality
@@ -1,20 +1,29 @@
 # LogicTest: 3node-tenant-multiregion
 skip under duress 118627
 
+user host-cluster-root
+
+statement ok
+ALTER TENANT [10] GRANT CAPABILITY can_admin_relocate_range=true
+
+user root
+
 # Create a table on the secondary tenant.
 statement ok
 CREATE TABLE t (k INT PRIMARY KEY, v INT, FAMILY (k, v))
+
+# Upreplicate the table's range.
+statement ok
+ALTER TABLE t EXPERIMENTAL_RELOCATE VALUES (ARRAY[1, 2, 3], 0)
 
 # Split the ranges in the table.
 statement ok
 ALTER TABLE t SPLIT AT SELECT generate_series(1, 6)
 
-# Relocate ranges in the admin tenant so that
+# Relocate ranges so that
 # - [1-2) and [2-3) are on node 2
 # - [3-4) and [4-5) are on node 3
 # - [5-6) and [6-7) are on node 1.
-user host-cluster-root
-
 statement ok
 ALTER RANGE RELOCATE LEASE TO 2 FOR SELECT range_id FROM crdb_internal.ranges WHERE start_pretty LIKE '%Tenant%1';
 ALTER RANGE RELOCATE LEASE TO 2 FOR SELECT range_id FROM crdb_internal.ranges WHERE start_pretty LIKE '%Tenant%2';
@@ -23,7 +32,7 @@ ALTER RANGE RELOCATE LEASE TO 3 FOR SELECT range_id FROM crdb_internal.ranges WH
 ALTER RANGE RELOCATE LEASE TO 1 FOR SELECT range_id FROM crdb_internal.ranges WHERE start_pretty LIKE '%Tenant%5';
 ALTER RANGE RELOCATE LEASE TO 1 FOR SELECT range_id FROM crdb_internal.ranges WHERE start_pretty LIKE '%Tenant%6';
 
-# Check range lease holders in the admin tenant.
+# Check range lease holders.
 query TI rowsort,retry
 SELECT start_pretty, lease_holder FROM crdb_internal.ranges WHERE start_pretty LIKE '%Tenant%Table%/1/%'
 ----
@@ -33,8 +42,6 @@ SELECT start_pretty, lease_holder FROM crdb_internal.ranges WHERE start_pretty L
 /Tenant/10/Table/106/1/4  3
 /Tenant/10/Table/106/1/5  1
 /Tenant/10/Table/106/1/6  1
-
-user root
 
 # Populate the range cache.
 statement ok

